integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -26,6 +26,7 @@
 
 
 DEFAULT_STARTER_PLAN_SPEC = {"serverless": {"region": "us-east-1", "cloud": "aws"}}
+METADATA_SUPPORTED_TYPES = str, int, bool  # List[str] is supported and checked separately
```

Comment:
it seems that also `float` is supported
https://docs.pinecone.io/guides/data/filter-with-metadata#supported-metadata-types